pkg/storageincentives/redistribution/redistribution.go

```diff
@@ -186,7 +189,7 @@ func (c *contract) sendAndWait(ctx context.Context, request *transaction.TxReque
 	if err != nil {
 		return err
 	}
-
+	c.setFee(c.txService.GetAccumulativeFee())
```

Comment:
like I wrote, this is not the right way to do it.

instead, we should have a new function in the transaction service that would return the transaction fee based on a transaction hash. 

and here you would pass the `txHash` returned from `txService.Send` to this new `TransactionFee` function.

how the transaction fee is acquired using the transaction hash requires a bit of research 
@swarmHaseeb @metacertain @vladopajic your input is welcome here